### ISSUE TITLE
mmgg.feeder.fi1 not working in new version

### ISSUE THREAD

**Issue by isaemirdag** [al-one/hass-xiaomi-miot#2003]
### Device model / 设备型号

mmgg.feeder.fi1

### Component version / 插件版本

1.0.1

### HA core version / HA版本

2024.12.0b4

### Integrated mode / 集成方式

Token

### The problem / 问题详情

Bugün Xiaomi Miot Auto için yeni versiyon update geldi. Update ettikten sonra mmgg.feeder.fi1 için yem durumu artık HA üzerinde görülmüyor ve kullanılamıyor. Cihazı silip yeniden kurdum fakat durum değişmedi. Yardım edermisiniz

### Entity attributes / 实体属性

```yaml
Unfortunately it is no longer visible
```

**Comment by Azaelyus** [comment]
I can confirm, my automation to serve X portions has stopped working:
![image](https://github.com/user-attachments/assets/bd5193dc-31a3-4990-af8d-9179248ef546)

I used to call the action on the sensor Food Level, which contained all information about the feeder - this seems to be replaced by the Info entity now?

**Comment by z-master42** [comment]
That's right, all attributes are now on the new info entity (button). The food level entity no longer exists. However, the actions cannot be used with the new button entity.

**Comment by Azaelyus** [comment]
So what is the workaround now? How can we dispense X portions? How can we see the food level?

**Comment by Azaelyus** [comment]
I managed to add the entity Food Level via customizing entity properties of the feeder:
![image](https://github.com/user-attachments/assets/52e05bdb-15a5-48d7-af1c-4fda53831e4e)
sensor_properties:: pet_food_left_level

This now works just fine. IMHO this property should be added by default now.

Still dunno how to dispense X portions.

**Comment by Azaelyus** [comment]
This seems to be working:

action: xiaomi_miot.call_action
data:
  entity_id: button.frufidr_pet_food_out
  siid: 2
  aiid: 1
  params: X (number of portions)

**Comment by z-master42** [comment]
> I managed to add the entity Food Level via customizing entity properties of the feeder:
> ![image](https://github.com/user-attachments/assets/52e05bdb-15a5-48d7-af1c-4fda53831e4e)
> sensor_properties:: pet_food_left_level
> 
> This now works just fine. IMHO this property should be added by default now.
> 
> Still dunno how to dispense X portions.

Which entity did you use for this?

> This seems to be working:
> 
> action: xiaomi_miot.call_action
> data:
> entity_id: button.frufidr_pet_food_out
> siid: 2
> aiid: 1
> params: X (number of portions)

I'll give that a try. It didn't work for me with the info entity, as far as I can remember.

I am currently back on 0.7.25.

**Comment by Azaelyus** [comment]
You have to customize the device and add the following string "pet_food_left_level" under "sensor_properties" field.
<img width="755" alt="image" src="https://github.com/user-attachments/assets/c7d28d53-1787-4fba-89be-112abeb2c6ee">

For dispensing X amount of portions you have to use the entity Pet Food out (you have to use xiaomi_miot_call_action function above). This entity can also be added via adding string "pet_food_out" under "button_actions" field.
![image](https://github.com/user-attachments/assets/52534b38-bda9-4dba-b533-e9bc2dd01af5)

Please remember to restart your HA instance once you do the changes above so the entities get populated.

**Comment by alxstojanovic** [comment]
The error reporting through the entity sensor.mmgg_fi1_XXXX_pet_feeder_2 has stipped working. HA says that the entity is no longer being provided. Why was it dropped?

**Comment by isaemirdag** [comment]
> You have to customize the device and add the following string "pet_food_left_level" under "sensor_properties" field. <img alt="image" width="755" src="https://private-user-images.githubusercontent.com/140115406/392327627-c7d28d53-1787-4fba-89be-112abeb2c6ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMzNDA0NzMsIm5iZiI6MTczMzM0MDE3MywicGF0aCI6Ii8xNDAxMTU0MDYvMzkyMzI3NjI3LWM3ZDI4ZDUzLTE3ODctNGZiYS04OWJlLTExMmFiZWIyYzZlZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNFQxOTIyNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNWNhYTY3NTcxOWMwOWE0MzJiZDMzOGM2MjlkOTcwNTAzMzNhMWQ3MWE2MDY5NzQwY2M4YzQxZmE1N2ZlZjJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6XgDl296awvwFuCa8dZkPRxS8DBsfsSj8qELQpVjgq8">
> 
> For dispensing X amount of portions you have to use the entity Pet Food out (you have to use xiaomi_miot_call_action function above). This entity can also be added via adding string "pet_food_out" under "button_actions" field. ![image](https://private-user-images.githubusercontent.com/140115406/392328394-52534b38-bda9-4dba-b533-e9bc2dd01af5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMzNDA0NzMsIm5iZiI6MTczMzM0MDE3MywicGF0aCI6Ii8xNDAxMTU0MDYvMzkyMzI4Mzk0LTUyNTM0YjM4LWJkYTktNGRiYS1iNTMzLWU5YmMyZGQwMWFmNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIwNFQxOTIyNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wOWVmNGFhMDhlMzMxYmE1NmRlMWM5ZmY1ODY3YmIzMmJkMjA5OTY5YzEzYWQyMGU4NjRiNmEyNzRjOTllYTZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.xRr3roFOASgU1k-oMrjZMkJQDFVk5u0snsOr45BI388)
> 
> Please remember to restart your HA instance once you do the changes above so the entities get populated.

problem solved thank you very much

**Comment by al-one** [comment]
Try master branch.
```bash
wget -O - https://get.hacs.vip | DOMAIN=xiaomi_miot ARCHIVE_TAG=master bash -
```
